src/test/groovy/nu/studer/gradle/buildscan/teamcity/TeamCityBuildScanPluginTest.groovy

```diff
@@ -51,32 +54,121 @@ class TeamCityBuildScanPluginTest extends Specification {
         buildScript = projectDir.newFile("build.gradle")
         runner = GradleRunner.create()
                 .withProjectDir(projectDir.root)
-                .withPluginClasspath()
     }
 
-    def "service messages emitted for compatible plugin versions"() {
+    def "service messages emitted for compatible build scan plugin versions"() {
         given:
-        applyScanPlugin("1.16")
+        configureBuildScanPlugin(BUILD_SCAN_PLUGIN_CLASSPATH)
+        buildScript << """
+            plugins {
+                id 'com.gradle.build-scan' version '1.16'
+                id 'nu.studer.build-scan.teamcity'
+            }
+        """.stripIndent()
+        configureScanPlugin()
 
         when:
         def result = runner.withArguments("tasks", "-S").build()
 
         then:
-        result.output.contains("##teamcity[nu.studer.teamcity.buildscan.buildScanLifeCycle 'BUILD_SCAN_URL:${mockScansServer.address}s/${PUBLIC_SCAN_ID}'")
+        outputContainsExpectedMessage(result.output)
     }
 
-    private void applyScanPlugin(String version) {
-        buildScript.text = """
+    def "service messages emitted when build scan plugin is applied later than the plugin"() {
+        given:
+        configureBuildScanPlugin(BUILD_SCAN_PLUGIN_CLASSPATH)
+        buildScript << """
             plugins {
-                id 'com.gradle.build-scan' version '${version}'
                 id 'nu.studer.build-scan.teamcity'
+                id 'com.gradle.build-scan' version '1.16'
             }
-            
+        """.stripIndent()
+        configureScanPlugin()
+
+        when:
+        def result = runner.withArguments("tasks", "-S").build()
+
+        then:
+        outputContainsExpectedMessage(result.output)
+    }
+
+    def "service messages emitted for compatible Gradle Enterprise plugin versions"() {
+        given:
+        runner.withGradleVersion("6.0-rc-1") // Gradle 6.0 is required for GE plugin 3.0
```

Comment:
@facewindu This can be updated to Gradle 6.0 now.

Reply:
done